File: app/src/main/java/org/ole/planet/myplanet/model/RealmMyPersonal.java
package org.ole.planet.myplanet.model;

import android.content.Context;

import com.google.gson.Gson;
import com.google.gson.JsonObject;

import org.ole.planet.myplanet.utilities.FileUtils;
import org.ole.planet.myplanet.utilities.NetworkUtils;

import java.util.Date;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

public class RealmMyPersonal extends RealmObject {
    @PrimaryKey
    private String id;
    private String _id;
    private String _rev;
    private boolean uploaded;
    private String title;
    private String description;
    private long date;
    private String userId;
    private String userName;
    private String path;


    public static JsonObject serialize(RealmMyPersonal personal, Context context) {
        JsonObject object = new JsonObject();
        object.addProperty("title", personal.getTitle());
        object.addProperty("uploadDate", new Date().getTime());
        object.addProperty("createdDate", personal.getDate());
        object.addProperty("filename", FileUtils.getFileNameFromUrl(personal.getPath()));
        object.addProperty("author", personal.getUserName());
        object.addProperty("addedBy", personal.getUserName());
        object.addProperty("description", personal.getDescription());
        object.addProperty("resourceType", "Activities");
        object.addProperty("private", true);
        JsonObject object1 = new JsonObject();
        object.addProperty("androidId", NetworkUtils.getMacAddr());
        object.addProperty("deviceName", NetworkUtils.getDeviceName());
        object.addProperty("customDeviceName", NetworkUtils.getCustomDeviceName(context));
        object1.addProperty("users", personal.getUserId());
        object.add("privateFor", object1);
//        object.addProperty("mediaType", FileUtils.getMediaType(personal.getPath()));
        return object;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public void set_rev(String _rev) {
        this._rev = _rev;
    }

    public boolean isUploaded() {
        return uploaded;
    }

    public void setUploaded(boolean uploaded) {
        this.uploaded = uploaded;
    }

    public String get_id() {
        return _id;
    }

    public void set_id(String _id) {
        this._id = _id;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String get_rev() {
        return _rev;
    }

    public String getTitle() {
        return title;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public long getDate() {
        return date;
    }

    public void setDate(long date) {
        this.date = date;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }
}

File: app/src/main/java/org/ole/planet/myplanet/model/RealmMyPersonal.kt
package org.ole.planet.myplanet.model

import android.content.Context
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import com.google.gson.JsonObject
import java.util.Date
import org.ole.planet.myplanet.utils.FileUtils
import org.ole.planet.myplanet.utils.NetworkUtils

/**
 * Room replacement for the former Realm `RealmMyPersonal` model. The class name is kept because
 * the UI and upload path use it purely as a detached data holder. Persistence goes through
 * [org.ole.planet.myplanet.data.room.dao.PersonalDao].
 */
@Entity(tableName = "my_personal", indices = [Index("userId")])
open class RealmMyPersonal {
    // @JvmField (field access, no generated getters) so Room does not treat the local `id` and the
    // CouchDB `_id` as ambiguous accessors (getId vs get_id both normalise to "id").
    @PrimaryKey
    @JvmField
    var id: String = ""
    @JvmField
    var _id: String? = null
    var _rev: String? = null
    var isUploaded = false
    var title: String? = null
    var description: String? = null
    var date: Long = 0
    var userId: String? = null
    var userName: String? = null
    var path: String? = null

    companion object {
        fun serialize(personal: RealmMyPersonal, context: Context): JsonObject {
            val `object` = JsonObject()
            `object`.addProperty("title", personal.title)
            `object`.addProperty("uploadDate", Date().time)
            `object`.addProperty("createdDate", personal.date)
            `object`.addProperty("filename", FileUtils.getFileNameFromUrl(personal.path))
            `object`.addProperty("author", personal.userName)
            `object`.addProperty("addedBy", personal.userName)
            `object`.addProperty("description", personal.description)
            `object`.addProperty("resourceType", "Activities")
            `object`.addProperty("private", true)
            val object1 = JsonObject()
            `object`.addProperty("androidId", NetworkUtils.getUniqueIdentifier())
            `object`.addProperty("deviceName", NetworkUtils.getDeviceName())
            `object`.addProperty("customDeviceName", NetworkUtils.getCustomDeviceName(context))
            object1.addProperty("users", personal.userId)
            `object`.add("privateFor", object1)
            return `object`
        }
    }
}
